src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -524,6 +524,20 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     GetEmitter()->emitIns_R_F(INS_fmov, emitTypeSize(intrin.baseType), targetReg, dataValue,
                                               INS_OPTS_NONE);
                 }
+                else if (intrin.op3->isContained() && intrin.op3->OperIs(GT_HWINTRINSIC))
```

Comment:
I agree with Tanner here - we should lower `Insert(dst, index1, src.GetElement(index2))`  and `InsertSelectedScalar(dst, index1, src, index2` (as we discussed recently) to `Insert(dst, index1, Extract(src, index2))` and contain its `op3` and here do as follows:
```c++
 }
else if (intrin.op3->isContained())
{
  assert(intrin.op3->OperIs(GT_HWINTRINSIC));
  assert(intrin.op3->AsHWIntrinsic()->gtHWIntrinsicId == AdvSimd_Extract);
  // emit mov dstReg[index1], srcReg[index2]
```
}

Reply:
> Rather than special casing this in multiple places, 

Could you confirm which places are you referring? Currently, in lower, we detect this pattern and mark the `op3` as contained and then here in code gen, emit the right instruction. 

> to Insert(dst, index1, Extract(src, index2)

We can't use `AdvSimd.Extract()` because it doesn't support `Vector64<long>`, `Vector64<ulong>` or `Vector64<double>` hence I am using `GetElement(0)`. Could you elaborate your suggestion. (Ignore `InsertSelectedScalar` for now which will be done in separate PR).